### PR TITLE
#165273439 Users should be able to rate articles and view article ratings

### PR DESCRIPTION
**Pull request by joelmugaya**
#### What does this PR do?
Enables a user to rate articles and view rating of articles

#### Description of Task to be completed
- create a reusable rating component
- display average rating of an article
- add clickable stars for rating
- have real time ratings update

#### How should this be manually tested?
Navigate to `/articles/<article_id>`

#### What are the relevant pivotal tracker stories?
[#165273439](https://www.pivotaltracker.com/story/show/165273439)

#### Screenshots (if appropriate)
Screenshot 2019-06-13 at 15 51 16

### CODE REVIEW

File: src/app/containers/RateArticle.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+  }
+    onClick = value => {
+      const post = {
+        rating: value
+      };
+      this.props.rateArticleAction(post, this.props.articleId);
+    }
+    render () {
+      function Repeat (props) {
```
Comment (njeri-ngigi) [inline src/app/containers/RateArticle.jsx]: put this method outside render and use es6 for it

Comment (njeri-ngigi) [reply src/app/containers/RateArticle.jsx]: render should be a pure function that does only one thing (display components)

File: src/redux/actions/RateArticleAction.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { rateArticleTypes } from './types';
+import { APP_URLS } from '../../utilities';
+import { toast } from 'react-toastify';
+
+const { GET_ARTICLE_RATING, RATE_ARTICLE } = rateArticleTypes;
+
+const furtherHandling = (error) => {
+  if (error.error) {
+    if (error.error === 'Rating for article doesnot exists') {
+      // console.log(error.error);
```
Comment (njeri-ngigi) [inline src/redux/actions/RateArticleAction.js]: commented out code?

File: src/redux/actions/RateArticleAction.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  }
+};
+
+export const handleErrors = (err) => {
+  if (err.response !== undefined) {
+    const error = err.response.data;
+    if (error.detail) {toast.error(error.detail, 'error', 5000);}
+    furtherHandling(error);
+  }
+  // else { console.log(err.Error); }
```
Comment (njeri-ngigi) [inline src/redux/actions/RateArticleAction.js]: commented out code

File: src/redux/actions/RateArticleAction.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import axios from 'axios';
+import { rateArticleTypes } from './types';
+import { APP_URLS } from '../../utilities';
+import { toast } from 'react-toastify';
+
+const { GET_ARTICLE_RATING, RATE_ARTICLE } = rateArticleTypes;
```
Comment (njeri-ngigi) [inline src/redux/actions/RateArticleAction.js]: good use of destructuring

**Review by njeri-ngigi — DISMISSED** [review]

**Review by engjames — APPROVED** [review]

**Review by sanya-kenneth — APPROVED** [review]

**Review by KabohaJeanMark — APPROVED** [review]
Good work Joel. I like how you have implemented this. This has worked quite fine. You are following all conventions and best practices with good destructuring and working on feedback to implement arrow functions.

**Review by 3Nakajugo — APPROVED** [review]

**Pull request merged**